rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -418,20 +418,36 @@ ImageData ROSImageTexture::setFormatAndNormalizeDataIfNecessary(ImageData image_
     image_data.encoding_ == sensor_msgs::image_encodings::MONO16)
   {
     image_data.size_ /= sizeof(uint16_t);
+    if (!bufferptr_) {
+      bufferptr_ = std::make_shared<std::vector<uint8_t>>(image_data.size_);
+    } else if (static_cast<size_t>(bufferptr_->size()) != image_data.size_) {
+      bufferptr_->resize(image_data.size_, 0);
+    }
+
     std::vector<uint8_t> buffer = normalize<uint16_t>(
       reinterpret_cast<const uint16_t *>(image_data.data_ptr_),
       image_data.size_);
+    std::copy(buffer.begin(), buffer.end(), bufferptr_->begin());
```

Comment:
Do we need to do this copy?  That is, instead of normalizing into a temporary buffer and then copying into the `bufferptr_`, can we just normalize directly into `bufferptr_`?  That probably requires changes to `normalize` and `createNewNormalizedBuffer`, but it should be a lot more performant.

The same comment goes for the code below.

Reply:
fixed https://github.com/ros2/rviz/pull/726/commits/401b05200ec9f260349be1a04b947c0abc5db40d